test/test_shadowState.py

```diff
@@ -0,0 +1,39 @@
+from unittest import TestCase
+
+from awsiot.iotshadow import ShadowState
+
+
+class ShadowTest(TestCase):
+
+    def test_shadow_state_payload_filled(self):
```

Comment:
add a test that uses from_payload() on something with nulls in it, which is the bug this PR is trying to address:

payload_with_none = {"reported": None, "desired": {"Color": "Blue}}
test_state = ShadowState.from_payload(payload_with_none)
self.assertTrue(test_state.desired_is_nullable)
self.assertEqual(test_state.to_payload(), payload_with_none)

payload_without_none = {"desired": {"Color": "Blue}}
test_state = ShadowState.from_payload(payload_without_none)
self.assertFalse(test_state.desired_is_nullable)
self.assertEqual(test_state.to_payload(), payload_without_none)